toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -840,6 +919,84 @@ func TestMetaNodeAddPkg(t *testing.T) {
 	assert.Equal(t, 5, len(g.AllNodesFrom(c.RunNode)))
 }
 
+func TestGoalWithLevelOneAndMeta(t *testing.T) {
+	g, err := buildTestGraphHelper()
+	assert.NoError(t, err)
+	assert.NotNil(t, g)
+
+	c1, _ := g.FindBestPkgNode(&pkgC)
+	c2, _ := g.FindBestPkgNode(&pkgC2)
+	meta := g.AddMetaNode([]*PkgNode{c2.RunNode}, []*PkgNode{c1.RunNode})
+
+	goal, err := g.AddGoalNodeWithExtraLayers("test_1meta", []*pkgjson.PackageVer{&pkgC}, false, 1)
+	assert.NoError(t, err)
+	assert.NotNil(t, goal)
+	nodesInGoal := []*PkgNode{}
+	for _, n := range graph.NodesOf(g.From(goal.ID())) {
+		nodesInGoal = append(nodesInGoal, n.(*PkgNode))
+	}
+	expectedGoalNodes := []*PkgNode{
+		pkgCRun,
+		pkgC2Run,
+		pkgBRun,
+	}
+	checkEqualComponents(t, expectedGoalNodes, nodesInGoal)
+	// But we now pull in the entire graph when looking at the tree
+	expectedGoalPackages2meta := []*PkgNode{
+		pkgBRun,
+		pkgBBuild,
+		pkgCRun,
+		pkgCBuild,
+		pkgD2Unresolved,
+		pkgD3Unresolved,
+		pkgC2Run,
+		pkgC2Build,
+		pkgD4Unresolved,
+		pkgD5Unresolved,
+		pkgD6Unresolved,
+		meta,
+		goal,
+	}
+	checkEqualComponents(t, expectedGoalPackages2meta, g.AllNodesFrom(goal))
+}
+
+func TestGoalWithMultipleGoals(t *testing.T) {
```

Comment:
I think we're testing also the fact we're adding more layers here. Suggestion:
```suggestion
func TestGoalWithMultipleGoalsAndOneExtraLayer(t *testing.T) {
```